src/datasource.ts

```diff
@@ -221,15 +221,21 @@ export default class CogniteDatasource {
         const aggregation = response.config.data.aggregates;
         const aggregationPrefix = aggregation ? `${aggregation} ` : '';
         return datapoints.concat(
-          response.data.data.items.map(item => ({
-            target: labels[count++] ? labels[count - 1] : aggregationPrefix + item.name,
-            datapoints: item.datapoints
-              .filter(d => d.timestamp >= timeFrom && d.timestamp <= timeTo)
-              .map(d => {
-                const val = Utils.getDatasourceValueString(response.config.data.aggregates);
-                return [d[val] === undefined ? d.value : d[val], d.timestamp];
-              }),
-          }))
+          response.data.data.items.map(item => {
+            if (item.datapoints.length >= response.config.data.limit) {
```

Comment:
Is this per request? Don't we have to sum of the count of all the limits instead of each individual ts?

Reply:
This checks if any of the timeseries returned from `/dataquery` hits its limit.

For example, if you send a request for ten timeseries with aggregates, then each timeseries gets a limit of 10000/10 = 1000. Then if any of these ten timeseries return 1000 datapoints, then they probably didn't return all the datapoints for that timerange - so we would show this warning

Reply:
The whole point of this warning is just to warn users that they may not have gotten all the datapoints for their selected time range and to remove any confusion as to why their timeseries just ends abruptly in the middle

Again, this is just a quick and easy solution to this issue, we can improve on this later by pulling in all the datapoints for a time range - but that currently has a bunch of issues with limits and knowing where to start the next `/dataquery` call from